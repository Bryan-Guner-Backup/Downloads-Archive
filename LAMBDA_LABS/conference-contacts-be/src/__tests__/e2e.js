// import our production apollo-server instance
const { server } = require('../');
const gql = require('graphql-tag');

const { startTestServer, toPromise } = require('./__utils');

const GET_USER = gql`
  query UserList {
    user {
      name
      profile {
        type
        value
        privacy
      }
    }
  }
`;

describe('Server - e2e', () => {
  let stop, graphql;

  beforeEach(async () => {
    const testServer = await startTestServer(server);
    stop = testServer.stop;
    graphql = testServer.graphql;
  });

  afterEach(() => stop());

  it('gets user profile', async () => {
    const res = await toPromise(
      graphql({
        query: GET_USER
      })
    );

    expect(res).toMatchSnapshot();
  });
});
