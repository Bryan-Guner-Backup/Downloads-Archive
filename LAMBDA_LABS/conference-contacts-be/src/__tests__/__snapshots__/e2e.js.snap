// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e gets user profile 1`] = `
Object {
  "data": Object {
    "user": Object {
      "name": "Jonathan Picazo",
      "profile": Array [
        Object {
          "privacy": "PRIVATE",
          "type": "EMAIL",
          "value": "macbethjonathan@gmail.com",
        },
      ],
    },
  },
}
`;
